pkg/subnet/subnet.go

```diff
@@ -45,6 +125,38 @@ func FindAvailableClassC() (*net.IPNet, error) {
 	return nil, errors.New("no available CIDR")
 }
 
+func Available(subnet *net.IPNet) (bool, error) {
+	addrs, err := interfaceAddrs()
+	if err != nil {
+		return false, fmt.Errorf("failed to obtain interface addresses: %v", err)
```

Comment:
Use `%w` instead of `%v` when wrapping errors (I think there are a couple of other places too).